Source/DafnyLanguageServer.Test/Various/CompilationStatusNotificationTest.cs

```diff
@@ -96,23 +99,20 @@ method Abs(x: int) returns (y: int)
 ".TrimStart();
       var documentItem = CreateTestDocument(source);
       await client.OpenDocumentAndWaitAsync(documentItem, CancellationToken);
-      var compilation = await notificationReceiver.AwaitNextNotificationAsync(CancellationToken);
-      Assert.AreEqual(documentItem.Uri, compilation.Uri);
-      Assert.AreEqual(documentItem.Version, compilation.Version);
-      Assert.AreEqual(CompilationStatus.CompilationSucceeded, compilation.Status);
-      var started = await notificationReceiver.AwaitNextNotificationAsync(CancellationToken);
-      Assert.AreEqual(documentItem.Uri, started.Uri);
-      Assert.AreEqual(documentItem.Version, started.Version);
-      Assert.AreEqual(CompilationStatus.VerificationStarted, started.Status);
-      var inprogress = await notificationReceiver.AwaitNextNotificationAsync(CancellationToken);
-      Assert.AreEqual(documentItem.Uri, inprogress.Uri);
-      Assert.AreEqual(documentItem.Version, inprogress.Version);
-      Assert.AreEqual(CompilationStatus.VerificationStarted, inprogress.Status);
-      Assert.AreEqual("Abs", inprogress.Message);
-      var completed = await notificationReceiver.AwaitNextNotificationAsync(CancellationToken);
-      Assert.AreEqual(documentItem.Uri, completed.Uri);
-      Assert.AreEqual(documentItem.Version, completed.Version);
-      Assert.AreEqual(CompilationStatus.VerificationSucceeded, completed.Status);
+      await AssertProgress(documentItem, CompilationStatus.CompilationSucceeded);
+      await AssertProgress(documentItem, CompilationStatus.VerificationStarted);
+      await AssertProgress(documentItem, CompilationStatus.VerificationStarted, "0/1 Abs");
+      await AssertProgress(documentItem, CompilationStatus.VerificationStarted, "1/1 (Abs finished)");
```

Comment:
Note the refactoring to make it easier to understand what the test does. Furthermore, now there are more messages because of real-time indications on how many methods need to be verified, the total number of methods, and what was the last verified method.